protocols/v2/binary-sv2/no-serde-sv2/codec/src/codec/decodable.rs

```diff
@@ -10,13 +10,35 @@ use std::convert::TryFrom;
 #[cfg(not(feature = "no_std"))]
```

Comment:
Can you check the lib.rs file. I have put that over there. The reason is that this module is being use inside this `pub mod decodable {
    pub use crate::codec::decodable::{Decodable, DecodableField, FieldMarker};
    //pub use crate::codec::decodable::PrimitiveMarker;
}`
Thus suppressing the top level docs to be shown..

Reply:
let's move these headers from `codec/src/lib.rs` to the respective files

that can be more intuitive for developer eyes

Reply:
I agree this feels a bit counterinuitive, but I tried moving these comments to the respective module files, but then `cargo doc` breaks with:

```
 cargo doc
   Compiling binary_codec_sv2 v1.2.0 (/Users/plebhash/develop/stratum/protocols/v2/binary-sv2/no-serde-sv2/codec)
 Documenting binary_codec_sv2 v1.2.0 (/Users/plebhash/develop/stratum/protocols/v2/binary-sv2/no-serde-sv2/codec)
error: missing documentation for a module
   --> v2/binary-sv2/no-serde-sv2/codec/src/lib.rs:145:1
    |
145 | pub mod encodable {
    | ^^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> v2/binary-sv2/no-serde-sv2/codec/src/lib.rs:1:9
    |
1   | #![deny(missing_docs)]
    |         ^^^^^^^^^^^^
```

Reply:
> I agree this feels a bit counterinuitive, but I tried moving these comments to the respective module files, but then `cargo doc` breaks with:
> 
> ```
>  cargo doc
>    Compiling binary_codec_sv2 v1.2.0 (/Users/plebhash/develop/stratum/protocols/v2/binary-sv2/no-serde-sv2/codec)
>  Documenting binary_codec_sv2 v1.2.0 (/Users/plebhash/develop/stratum/protocols/v2/binary-sv2/no-serde-sv2/codec)
> error: missing documentation for a module
>    --> v2/binary-sv2/no-serde-sv2/codec/src/lib.rs:145:1
>     |
> 145 | pub mod encodable {
>     | ^^^^^^^^^^^^^^^^^
>     |
> note: the lint level is defined here
>    --> v2/binary-sv2/no-serde-sv2/codec/src/lib.rs:1:9
>     |
> 1   | #![deny(missing_docs)]
>     |         ^^^^^^^^^^^^
> ```

Because I have added  `#![deny(missing_docs)]` macro to error in case of missing doc. As you moved the doc from public exposed mod to non-public exposed mod, it broke.